bytecode-verificator/bytecode_verificator.sh

```diff
@@ -0,0 +1,479 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+if [[ "${TRACE-0}" == "1" ]]; then
+    set -o xtrace
+fi
+
+################################
+# Bytecode verification script #
+################################
+
+RED='\033[0;31m'
+ORANGE='\033[0;33m'
+GREEN='\033[0;32m'
+NC='\033[0m'
+
+zero_padding=$(printf '%0.1s' "0"{1..64})
+placeholder_padding=$(printf '%0.1s' "-"{1..64})
+
+# Prerequisite executables
+prerequisites=(jq yarn awk curl shasum uname bc)
+
+# Environment vailable required
+envs=(WEB3_INFURA_PROJECT_ID ETHERSCAN_TOKEN)
+
+# Commandline args required
+cmdargs=(solc_version remote_rpc_url contract config_json)
+
+# Vars
+constructor_calldata=""
+contract_config_name=""
+local_rpc_url=""
+
+function show_help() {
+    echo -e "$ORANGE Usage: $NC $0"\
+      "--solc-version 0.x.y" \
+      "--remote-rpc https://*" \
+      "--config-json ../deployed-*.json" \
+      "--contract Name" \
+      "[--constructor-calldata *]"\
+      "[--proxy]"\
+      "[--implementation]"
+}
+
+# Fork PID of Ganache
+fork_pid=0
+
+# Entry point
+main() {
+  check_prerequisites
+  check_envs
+
+  parse_cmd_args "$@"
+  check_compiler
+  start_fork
+
+  compile_contract
+  deploy_contract_on_fork
+  compare_bytecode
+}
+
+# Service functions
+
+function check_prerequisites() {
+  for p in "${prerequisites[@]}"
+  do
+    if ! [ -x "$(command -v "$p")" ]; then
+      _err "$p app is required but not found"
+    fi
+  done
+}
+
+function check_envs() {
+  set +u
+  for e in "${envs[@]}"
+  do
+    if [[ -z "${!e}" ]]; then
+      _err "${e} env var is required but is not set"
+    fi
+  done
+  set -u
+}
+
+function parse_cmd_args() {
+  while [[ $# -gt 0 ]]; do
+    case $1 in
+      --solc-version)
+        solc_version="$2"
+        shift
+        shift
+        ;;
+      --remote-rpc-url)
+        remote_rpc_url="$2"
+        shift
+        shift
+        ;;
+      --contract)
+        contract="$2"
+        shift
+        shift
+        ;;
+      --config-json)
+        config_json="$2"
+        shift
+        shift
+        ;;
+      --constructor-calldata)
+        constructor_calldata="$2"
+        shift
+        shift
+        ;;
+      --proxy)
+        is_proxy=true
+        shift
+        ;;
+      --implementation)
+        is_implementation=true
+        shift
+        ;;
+      --help)
+        show_help
+        exit 0
+        ;;
+      --*|-*)
+        _err "Unknown option \"$1\""
+        ;;
+    esac
+  done
+
+  for arg in "${cmdargs[@]}"
+  do
+    if [ -z "${!arg}" ]; then
+      _err "argument '--${arg//_/-}' is empty"
+    fi
+  done
+}
+
+function check_compiler() {
+  platform=$(uname | awk '{print tolower($0)}')
+  sha256sum='sha256sum'
+  if [[ $platform == 'darwin' ]] || [[ $platform == 'linux' ]]; then
+    sha256sum='shasum -a 256'
+  else
+    _err "unknown platform: $platform"
+  fi
```

Comment:
Redundant? We have the hardcoded command in `_print_checksum` function.
The platform check can be extracted to the separate function.